app/services/fetch_web_document.rb

```diff
@@ -28,7 +28,15 @@ def set_document_attachment
   end
 
   def fetch_file_content
-    content = HTTParty.get(@document.link)
+    begin
+      content = HTTParty.get(@document.link)
+    rescue StandardError => e
+      set_document_attachment
+      tempfile.close
+      @document.update!(state: :errored, error_message: e.message)
```

Comment:
FetchWebDocumentJob should be doing this when the job retries are exhausted. Are you finding that that doesn't work?

Handling the errors in the job class is a bit of a pattern, because the job knows when it's giving up, whereas the service run by the job doesn't.

I don't even think you need to close the tempfile in the error case, since Tempfile will close the file description when its last reference is garbage collected.

Reply:
I think that since there's a definite error, it needs to error out here.  And if you don't close out the file it doesn't get attached and bombs out if anyone tries to view it.    
Maybe the `rescue` can be in the job, but you'd still want a blank file I think.  Otherwise you'd have to check for blank everywhere.

Reply:
Since the file fetching moved to a job, there will always be a time when a document can have a nil `file`, so some checking is going to have to happen, but you're right that we don't want to do it everywhere.

Maybe a method in Document itself would be the place to do this. It would be better not to have all of its collaborators doing `document.file.blob` and such anyway.